tools/src/main/scala/com/nvidia/spark/rapids/tool/profiling/AutoTuner.scala

```diff
@@ -339,157 +572,226 @@ class AutoTuner(app: ApplicationSummaryInfo, workerInfo: String) extends Logging
    *     Output: newMaxPartitionBytes = 2g / (512m/128m) = 512m
    */
   private def calculateMaxPartitionBytes(maxPartitionBytes: String): String = {
-    val taskInputSize =
-      app.sqlTaskAggMetrics.map(_.inputBytesReadAvg).sum / app.sqlTaskAggMetrics.size
-    val maxPartitionBytesNum = convertFromHumanReadableSize(maxPartitionBytes)
-    if (taskInputSize > 0 &&
-      taskInputSize < convertFromHumanReadableSize(MIN_PARTITION_BYTES_RANGE)) {
-      // Increase partition size
-      val calculatedMaxPartitionBytes = Math.min(
-        maxPartitionBytesNum *
-          (convertFromHumanReadableSize(MIN_PARTITION_BYTES_RANGE) / taskInputSize),
-        convertFromHumanReadableSize(MAX_PARTITION_BYTES_BOUND))
-
-      convertToHumanReadableSize(calculatedMaxPartitionBytes.toLong)
-    } else if (taskInputSize > convertFromHumanReadableSize(MAX_PARTITION_BYTES_RANGE)) {
-      // Decrease partition size
-      val calculatedMaxPartitionBytes = Math.min(
-        maxPartitionBytesNum /
-          (taskInputSize / convertFromHumanReadableSize(MAX_PARTITION_BYTES_RANGE)),
-        convertFromHumanReadableSize(MAX_PARTITION_BYTES_BOUND))
-
-      convertToHumanReadableSize(calculatedMaxPartitionBytes.toLong)
+    val app = appInfo.get
+    val maxPartitionBytesNum = convertToMB(maxPartitionBytes)
+    if (app.sqlTaskAggMetrics.isEmpty) { // avoid division by zero
+      maxPartitionBytesNum.toString
     } else {
-      // Do not recommend maxPartitionBytes
-      null
+      val taskInputSizeInMB =
+        (app.sqlTaskAggMetrics.map(_.inputBytesReadAvg).sum / app.sqlTaskAggMetrics.size) / 1024
```

Comment:
this is missing a divide by 1024 because we need to change bytes to MB... ie should be / 1024 / 1024